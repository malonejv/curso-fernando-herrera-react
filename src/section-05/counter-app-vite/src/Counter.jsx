import PropTypes from "prop-types";
import { useState } from "react";

const Counter = ({ count }) => {
  const [currentCount, setCurrentCount] = useState(count);
  const increment = () => {
    setCurrentCount(currentCount + 1);
  };

  const decrement = () => {
    setCurrentCount(currentCount - 1);
  };

  const reset = () => {
    setCurrentCount(count);
  };

  return (
    <div>
      <h1>Counter</h1>
      <p>Current Count: {currentCount}</p>
      <button onClick={increment}>+</button>
      <button onClick={decrement}>-</button>
      <button onClick={reset}>Reset</button>
    </div>
  );
};

export default Counter;

Counter.propTypes = {
  count: PropTypes.number.isRequired,
};

Counter.defaultProps = {
  count: 0,
};
