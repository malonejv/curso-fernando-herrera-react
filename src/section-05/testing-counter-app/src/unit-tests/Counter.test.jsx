import { render, screen, fireEvent } from "@testing-library/react";
import Counter from "../Counter.jsx";

describe("Component: Counter", () => {

    const initCount = 1;

    test("should match snapshot", () => {
        const { container } = render(<Counter count={initCount} />);
        expect(container).toMatchSnapshot();
    });

    test("should have an initial count of 100", () => {
        render(<Counter count={initCount} />);
        expect(screen.getByText(`Current Count: ${initCount}`)).toBeTruthy();
    });

    test("should increment count when + button is clicked", () => {
        render(<Counter count={initCount} />);
        const incrementButton = screen.getByText("+");
        fireEvent.click(incrementButton);
        expect(screen.getByText(`Current Count: ${initCount + 1}`)).toBeTruthy();
    });

    test("should decrement count when - button is clicked", () => {
        render(<Counter count={initCount} />);
        const decrementButton = screen.getByText("-");
        fireEvent.click(decrementButton);
        expect(screen.getByText(`Current Count: ${initCount - 1}`)).toBeTruthy();
    });

    test("should reset count to initial value when Reset button is clicked", () => {
        render(<Counter count={initCount} />);
        const incrementButton = screen.getByText("+");
        fireEvent.click(incrementButton);
        const resetButton = screen.getByText("Reset");
        fireEvent.click(resetButton);
        expect(screen.getByText(`Current Count: ${initCount}`)).toBeTruthy();
    });

});