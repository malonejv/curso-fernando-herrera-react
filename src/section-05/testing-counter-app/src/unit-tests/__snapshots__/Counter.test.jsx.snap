// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Counter should match snapshot 1`] = `
<div>
  <div>
    <h1>
      Counter
    </h1>
    <p>
      Current Count: 
      1
    </p>
    <button>
      +
    </button>
    <button>
      -
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
